packages/arb-ts/src/lib/l1Bridge.ts

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright 2019-2020, Offchain Labs, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/* eslint-env node */
+'use strict'
+import { providers, Signer, BigNumber } from 'ethers'
+import { InboxFactory } from './abi/InboxFactory'
+import { EthERC20BridgeFactory } from './abi/EthERC20BridgeFactory'
+import { EthERC20Bridge } from './abi/EthERC20Bridge'
+import { Inbox } from './abi/Inbox'
+
+export class L1Bridge {
+  l1Provider: providers.JsonRpcProvider
+  l1Signer: Signer
+  inbox: Inbox
+  ethERC20Bridge: EthERC20Bridge
+  walletAddressCache?: string
+
+  constructor(
+    inboxAddress: string,
+    erc20BridgeAddress: string,
+    l1Provider: providers.JsonRpcProvider,
+    l1Signer: Signer
+  ) {
+    this.l1Provider = l1Provider
+    this.l1Signer = l1Signer
+
+    const l1SignerOrProvider = l1Signer || l1Provider
+    this.inbox =
+      l1SignerOrProvider &&
+      InboxFactory.connect(inboxAddress, l1SignerOrProvider)
+
+    this.ethERC20Bridge = EthERC20BridgeFactory.connect(
+      erc20BridgeAddress,
+      l1Signer
+    )
+  }
+
+  public async depositETH(value: BigNumber, destinationAddress?: string) {
+    if (!this.l1Signer || !this.inbox) {
+      throw new Error('No eth signer provider')
+    }
+    const address = destinationAddress || (await this.getWalletAddress())
+    return this.inbox.depositEth(address, {
+      value,
+    })
+  }
+
+  public async depositERC20(
```

Comment:
I think we want `PayableOverrides` for paying for L2 gas

Reply:
is including `value` not enough?